src/main/java/blusunrize/immersiveengineering/common/items/CoresampleItem.java

```diff
@@ -52,57 +63,76 @@ public CoresampleItem()
 	@OnlyIn(Dist.CLIENT)
 	public void addInformation(ItemStack stack, @Nullable World world, List<ITextComponent> list, ITooltipFlag flag)
 	{
-		DimensionChunkCoords coords = getCoords(stack);
+		getCoresampleInfo(stack, list, TextFormatting.GRAY, world, true, true);
+	}
+
+	public static void getCoresampleInfo(ItemStack coresample, List<ITextComponent> list, TextFormatting baseColor, @Nullable World world, boolean showYield, boolean showTimestamp)
+	{
+		if(coresample.getOrCreateTag().contains("coords"))
+			list.add(new TranslationTextComponent(Lib.DESC_INFO+"coresample.outdated"));
+
+		ColumnPos coords = getCoords(coresample);
 		if(coords!=null)
 		{
-			MineralMix mineral = getMix(stack);
-			if(mineral!=null)
-			{
-				String unloc = mineral.getTranslationKey();
-				String loc = I18n.format(unloc);
-				if(unloc.equals(loc))
-					loc = mineral.getPlainName();
-				list.add(new TranslationTextComponent(Lib.CHAT_INFO+"coresample.mineral", loc));
-			}
+			ListNBT nbtList = coresample.getOrCreateTag().getList("mineralInfo", NBT.TAG_COMPOUND);
```

Comment:
Why isn't this using `getMineralMixes`?

Reply:
Because I need the other values in the NBT tag. `getMineralMixes` returns only the mixes, but I need saturation and depletion here.